tasks/clean-build-service.yml

```diff
@@ -0,0 +1,59 @@
+---
+- name: Check if systemd timer definition is present
+  stat:
+    path: '/etc/systemd/system/build-{{ beacon_node_service_name}}.timer'
+  register: systemd_service_build_timer
+
+- name: Check if the service is running
+  systemd_service:
+    name: 'build-{{ beacon_node_service_name }}.timer'
+    state: 'stopped'
+  when: systemd_service_build_timer.stat.exists
+
+# Allow to run the beacon node if the binary is build from source 
+- name: Create a symlink with the build binary
+  file:
+    src: '{{ beacon_node_repo_path }}/build/nimbus_beacon_node'
+    dest: '{{ beacon_node_bin_path }}/nimbus_beacon_node'
+    state: link
+  when: beacon_node_update_build 
+
+- name: Check if systemd service definition is present
+  stat:
+    path: '/etc/systemd/system/build-{{ beacon_node_service_name}}.service'
+  register: systemd_service_build
+
+- name: Remove service definition
+  file:
+    path: '/etc/systemd/system/build-{{ beacon_node_service_name }}.service'
+    state: absent
+  when: systemd_service_build.stat.exists
```

Comment:
Another pointless check. Just do one task with `iter_items` for all three of those without any `stat`s.